x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/amazonbedrock/AmazonBedrockServiceTests.java

```diff
@@ -143,6 +151,144 @@ public void testParseRequestConfig_ThrowsUnsupportedModelType() throws IOExcepti
         }
     }
 
+    public void testGetConfiguration() throws IOException {
+        try (var service = createAmazonBedrockService()) {
+            String content = XContentHelper.stripWhitespace("""
+                {
+                     "provider": "amazonbedrock",
+                     "task_types": [
+                         "text_embedding",
+                         "completion"
+                     ],
+                     "configuration": {
```

Comment:
I think we may need to split up `configuration` into `service_settings` and `task_settings`? @YulNaumenko does that sound right?

For example, https://www.elastic.co/guide/en/elasticsearch/reference/current/infer-service-amazon-bedrock.html
Could be:
```
PUT _inference/completion/amazon_bedrock_completion
{
    "service": "amazonbedrock",
    "service_settings": {
        "access_key": "<aws_access_key>",
        "secret_key": "<aws_secret_key>",
        "region": "us-east-1",
        "provider": "amazontitan",
        "model": "amazon.titan-text-premier-v1:0"
    },
    "task_settings": {
        "temperature": 0.5,
        "top_p": 0.5
    }
}
```

Do we need to communicate back to the kibana connector? Or am I misunderstanding the configuration object?

Reply:
I believe this is a separate issue: https://github.com/elastic/ml-team/issues/1308. Is that correct @YulNaumenko?

Reply:
I think we can make it within the single request as @prwhelan suggested and that will help to avoid calling two separate APIs. I will need to change the Connector code to consume the new schema.
For example we can extend it by adding to each connector the thing I requested as a separate API:

taskTypes: [
    {
          "task_type": string,
          "configuration": {
              category?: string;
              default_value: string | number | boolean | null;
              depends_on: Dependency[];
              display: DisplayType;
              label: string;
              options?: SelectOption[];
              order?: number | null;
              placeholder?: string;
              required: boolean;
              sensitive: boolean;
              tooltip: string | null;
              type: FieldType;
              ui_restrictions: string[];
              validations: Validation[];
              value: string | number | boolean | null;
       }
    }
  ]